### PR TITLE
feat: wire-up thorchain lending Available Pools

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Because programmatic is better than hardcoded values - brings actual THOR data to the Available pools rows:

 - Available lending pools
 - Lending pools data

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

None - this is under flag

## Testing

- Data looks the same in "Available Pools" and in the actual /pool/:pool route

### Engineering

- :point_up:
 
### Operations

- :point_up:

## Screenshots (if applicable)

https://github.com/shapeshift/web/assets/17035424/598a4d75-7434-44a7-a8d7-418992d6339e

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5568** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5568" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5568?utm_source=stack-comment).

File: src/pages/Lending/AvailablePools.tsx [hunk 1]
```diff
@@ -11,14 +10,70 @@ import { HelperTooltip } from 'components/HelperTooltip/HelperTooltip'
...
+const LendingPoolButton = ({
+  asset,
+  onPoolClick: handlePoolClick,
+}: {
+  asset: Asset
+  onPoolClick: (assetId: AssetId) => void
+}) => {
+  const usePoolDataArgs = useMemo(() => ({ poolAssetId: asset.assetId }), [asset.assetId])
+  const { data: poolData, isLoading: isPoolDataLoading } = usePoolDataQuery(usePoolDataArgs)
+  console.log({ poolData })
```
Comment (gomesalexandre) [inline src/pages/Lending/AvailablePools.tsx:33]: oh noes

File: src/pages/Lending/AvailablePools.tsx [hunk 2]
```diff
@@ -11,14 +10,70 @@ import { HelperTooltip } from 'components/HelperTooltip/HelperTooltip'
...
+      display='grid'
+      gridTemplateColumns={lendingRowGrid}
+      columnGap={4}
+      alignItems='center'
+      textAlign='left'
+      py={4}
+      width='full'
+      height='auto'
+      color='text.base'
+      // eslint-disable-next-line react-memo/require-usememo
```
Comment (gomesalexandre) [inline src/pages/Lending/AvailablePools.tsx:46]: Now that this is its own component, we can memo the callback

File: src/pages/Lending/AvailablePools.tsx (hunk 1 above)
Comment (gomesalexandre) [reply src/pages/Lending/AvailablePools.tsx]: fe36ee6b1

File: src/pages/Lending/AvailablePools.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/pages/Lending/AvailablePools.tsx]: fe36ee6b1

File: src/pages/Lending/AvailablePools.tsx [hunk 3]
```diff
@@ -11,14 +10,70 @@ import { HelperTooltip } from 'components/HelperTooltip/HelperTooltip'
...
   md: '200px repeat(5, 1fr)',
 }
 
+const LendingPoolButton = ({
+  asset,
+  onPoolClick: handlePoolClick,
+}: {
+  asset: Asset
+  onPoolClick: (assetId: AssetId) => void
+}) => {
```
Comment (gomesalexandre) [inline src/pages/Lending/AvailablePools.tsx:30]:
```suggestion
}: LendingPoolButtonProps) => {
```

**Review by woodenfurniture — APPROVED** [review]

**Review by 0xApotheosis — APPROVED** [review]

**Pull request merged**